source/common/router/scoped_rds.cc

```diff
@@ -227,40 +228,40 @@ void ScopedRdsConfigSubscription::onConfigUpdate(
   // NOTE: deletes are done before adds/updates.
 
   absl::flat_hash_map<std::string, ScopedRouteInfoConstSharedPtr> to_be_removed_scopes;
+  // Destruction of resume_rds will lift the floodgate for new RDS subscriptions.
+  // Note in the case of partial acceptance, accepted RDS subscriptions should be started
+  // despite of any error.
+  ScopedResume resume_rds;
   // If new route config sources come after the local init manager's initialize() been
   // called, the init manager can't accept new targets. Instead we use a local override which will
   // start new subscriptions but not wait on them to be ready.
   std::unique_ptr<Init::ManagerImpl> noop_init_manager;
-  // NOTE: This should be defined after noop_init_manager as it depends on the
-  // noop_init_manager.
-  std::unique_ptr<Cleanup> resume_rds;
+  // NOTE: This should be defined after noop_init_manager and resume_rds, as it depends on the
+  // noop_init_manager, and we want a single RDS discovery request to be sent to management server.
+  std::unique_ptr<Cleanup> init_noop_init_manager;
```

Comment:
Can you give this a clearer name? It's easy to confuse `noop_init_manager` and `init_noop_init_manager` when reading.

Reply:
how about disposable_init_mgr?